tests/run-macros/tasty-construct-types/Macro_1.scala

```diff
@@ -0,0 +1,63 @@
+import tasty._
+import quoted._
+
+object Macros {
+  inline def theTestBlock : Unit = ${ theTestBlockImpl }
+
+  trait RefineMe {
+    type T
+    def foo : T
+  }
+
+  class TestAnnotation extends scala.annotation.Annotation
+
+  def theTestBlockImpl(given qctx : QuoteContext) : Expr[Unit] = {
+    import qctx.tasty.{_,given}
+    
+    val x1T = ConstantType(Constant(1)).seal.asInstanceOf[quoted.Type[Any]]
+    val x2T = OrType(ConstantType(Constant(1)), ConstantType(Constant(2))).seal.asInstanceOf[quoted.Type[Any]]
+    val x3T = AndType(ConstantType(Constant(3)), typeOf[Any]).seal.asInstanceOf[quoted.Type[Any]]
+    val x4Lam =
+      TypeLambda(
+        List("A","B"),
+        _ => List(TypeBounds(typeOf[Nothing], typeOf[Any]), TypeBounds(typeOf[Nothing], typeOf[Any])),
+        (tl : TypeLambda) => tl.param(1)).seal.asInstanceOf[quoted.Type[[A,B]=>>B]]
+    val x5T =
+      Refinement(
+        typeOf[RefineMe],
+        "T",
+        TypeBounds(typeOf[Int], typeOf[Int])).seal.asInstanceOf[quoted.Type[Any]]
+    val x6T = AppliedType(Type(classOf[List[_]]), List(typeOf[Int])).seal.asInstanceOf[quoted.Type[Any]]
+    val x7T = AnnotatedType(ConstantType(Constant(7)), '{ new TestAnnotation }.unseal).seal.asInstanceOf[quoted.Type[Any]]
+    val x8T =
+      MatchType(
+        typeOf[Int],
+        typeOf[List[8]],
+        List(
+          TypeLambda(
+            List("t"),
+            _ => List(TypeBounds(typeOf[Nothing], typeOf[Any])),
+            tl => AppliedType(MatchCaseType, List(AppliedType(Type(classOf[List[_]]), List(tl.param(0))), tl.param(0)))))
+      ).seal.asInstanceOf[quoted.Type[Any]]
+
+    '{
+      val x1 : ${x1T} = 1
```

Comment:
This is only testing that `1` is a subtype of `x1T` but not the other way around. 
I propose to try a more direct approach with assetrions and without `asInstanceOf` like:

```scala
val x1T = ConstantType(Constant(1))
val x2T = OrType(ConstantType(Constant(1)), ConstantType(Constant(2)))
val x3T = AndType(ConstantType(Constant(3)), typeOf[Any])
val x4Lam =
      TypeLambda(
        List("A","B"),
        _ => List(TypeBounds(typeOf[Nothing], typeOf[Any]), TypeBounds(typeOf[Nothing], typeOf[Any])),
        (tl : TypeLambda) => tl.param(1))
val x5T =
      Refinement(
        typeOf[RefineMe],
        "T",
        TypeBounds(typeOf[Int], typeOf[Int]))
...

assert(x1T =:= '[1].unseal.tpe)
assert(x2T =:= '[1 | 2].unseal.tpe)
assert(x3T =:= '[1 & Any].unseal.tpe)
assert(x4T =:= '[[A, B] =>> B].unseal.tpe)
assert(x5T =:= '[RefineMe { type T = Int }].unseal.tpe)
...
``` 

Then you could also move the test to `tests/run-macros`

Reply:
Good point. I've updated the tests to do this (code push coming soon).

I don't understand "move the test to `tests/run-macros`" though. That is already where the test is?

Reply:
I meant `tests/pos-macros`. Sorry.

Reply:
But it is ok like this